### PR TITLE
[IMP] various: improves usage of partners

### PR DESCRIPTION
**Pull request by pko-odoo**
SPECIFICATION

Various change for the partner view:
- Moved the activity widget to the bottom right of the kanban
- Moved the informations badge to the bottom right next to the
  activity widget and made them clickable
- Change the address options order and add a small help below
- Change the 'Remove' button function from delete to remove from
  the company and add a delete button to the right end.
- Correct some typo
- Add an 'Archived' ribbon to the kanban card
- Change various small things

LINKS

Task-2821356

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/legacy/js/views/res_partner_view_dialogs.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+/** @odoo-module alias=web.ResPartnerFormViewDialog **/
+
+import { _t } from 'web.core';
+import Dialog from 'web.Dialog';
+import FormViewDialog from 'web.view_dialogs';
```
Comment (std-odoo) [inline addons/web/static/src/legacy/js/views/res_partner_view_dialogs.js:5]: just to save some characters
```suggestion
import {FormViewDialog} from 'web.view_dialogs';
```

**Review by std-odoo — COMMENTED** [review]
@pko-odoo Hi :) Validation :) 

Code is clean I have not that much to say except some picky remarks :)

File: odoo/addons/base/models/res_partner.py [hunk 2]
```diff
@@ -1002,7 +1002,6 @@ def _get_country_name(self):
         return self.country_id.name or ''
 
 
-
```
Comment (std-odoo) [inline odoo/addons/base/models/res_partner.py:1005]: useless diff

**Review by std-odoo — COMMENTED** [review]
One last comment :)

File: odoo/addons/base/views/res_partner_views.xml [hunk 3]
```diff
@@ -452,21 +471,27 @@
...
+                                    <span class="bg-danger">Archived</span>
+                                </div>
+                                <div class="oe_kanban_details d-flex flex-column justify-content-between">
+                                    <div>
+                                        <strong class="o_kanban_record_title oe_partner_heading"><field name="display_name"/></strong>
+                                        <div class="o_kanban_tags_section oe_kanban_partner_categories"/>
+                                        <ul>
+                                            <li t-if="record.parent_id.raw_value and !record.function.raw_value"><field name="parent_id"/></li>
+                                            <li t-if="!record.parent_id.raw_value and record.function.raw_value"><field name="function"/></li>
+                                            <li t-if="record.parent_id.raw_value and record.function.raw_value"><field name="function"/> at <field name="parent_id"/></li>
```
Comment (Feyensv) [inline odoo/addons/base/views/res_partner_views.xml:484]:
```suggestion
                                            <li t-elif="!record.parent_id.raw_value and record.function.raw_value"><field name="function"/></li>
                                            <li t-elif="record.parent_id.raw_value and record.function.raw_value"><field name="function"/> at <field name="parent_id"/></li>
```
Little improvement while you're modifying the line :eyes:

File: addons/web/static/src/legacy/scss/res_partner.scss [hunk 4]
```diff
@@ -0,0 +1,29 @@
+.o_kanban_view.o_res_partner_kanban {
```
Comment (tde-banana-odoo) [inline addons/web/static/src/legacy/scss/res_partner.scss:1]: Shouldn't the css be in base, wiht the view definition ?

File: addons/web/static/src/legacy/js/views/res_partner_view_dialogs.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+import { _t } from 'web.core';
+import Dialog from 'web.Dialog';
+import { FormViewDialog } from 'web.view_dialogs';
+
+
+const ResPartnerFormViewDialog = FormViewDialog.extend({
+
+    _setRemoveButtonOption(options, btnClasses) {
+        options.buttons.push({
+            text: options.removeButtonText || _t("Remove"),
```
Comment (tde-banana-odoo) [inline addons/web/static/src/legacy/js/views/res_partner_view_dialogs.js:12]: That's strange that "remove" leads to breaking the parent-id herarchy. I don't think this is clearer than an unlink. Maybe to check with LNA but that seems really strange.

File: addons/web/static/src/legacy/js/views/res_partner_form_controller.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+import FormController from 'web.FormController';
+import ResPartnerFormViewDialog from 'web.ResPartnerFormViewDialog';
+
+
+const ResPartnerFormController = FormController.extend({
+
+    /**
+     * @override
+     */
+    _onOpenOne2ManyRecord: async function (ev) {
```
Comment (tde-banana-odoo) [inline addons/web/static/src/legacy/js/views/res_partner_form_controller.js:13]: What is the purpose of this code ? Aka: it lacks at least a docstring, as it is not crystal clear to me what it is used for :D

**Review by tde-banana-odoo — COMMENTED** [review]
Zboing !

File: addons/web/static/src/legacy/js/views/res_partner_view_dialogs.js (hunk 5 above)
Comment (pko-odoo) [reply addons/web/static/src/legacy/js/views/res_partner_view_dialogs.js:12]: Before this, when one clicked on the 'Remove' button from the FormViewDialog it actually deleted the partner...
LNA wanted a way to remove it from the company without deleting it. Maybe we should choose another wording ?

File: addons/web/static/src/legacy/scss/res_partner.scss (hunk 4 above)
Comment (pko-odoo) [reply addons/web/static/src/legacy/scss/res_partner.scss]: I actually hesitated about that... But I saw that the base module barely had any css files :smile:

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Pull request closed without merging**